src/Gui/Placement.h

```diff
@@ -80,49 +80,42 @@ private Q_SLOTS:
     bool changeProperty;
 };
 
-class GuiExport Placement : public QDialog
+
+class GuiExport PlacementBase : public QDialog
 {
     Q_OBJECT
 
+    friend class Placement;
+    friend class PlacementDialog;
+

```

Comment:
It's weird to make sub-classes a friend of a base class. This looks like some code-refactoring is needed to avoid it.